contracts/lib/LicensingHelper.sol

```diff
@@ -51,4 +49,61 @@ library LicensingHelper {
             }
         }
     }
+
+    /// @dev Registers PIL License Terms and attaches them to the given IP on behalf of the owner with a signature.
+    /// @param ipId The ID of the IP to which the license terms will be attached.
+    /// @param pilTemplate The address of the PIL License Template.
+    /// @param licensingModule The address of the Licensing Module.
+    /// @param terms The PIL terms to be registered.
+    /// @param sigAttach Signature data for attachLicenseTerms to the IP via the Licensing Module.
+    /// @return licenseTermsId The ID of the registered PIL terms.
+    function registerPILTermsAndAttachWithSig(
+        address ipId,
+        address pilTemplate,
+        address licensingModule,
+        PILTerms calldata terms,
+        WorkflowStructs.SignatureData calldata sigAttach
+    ) internal returns (uint256 licenseTermsId) {
+        licenseTermsId = IPILicenseTemplate(pilTemplate).registerLicenseTerms(terms);
+        attachLicenseTermsWithSig(ipId, licensingModule, pilTemplate, licenseTermsId, sigAttach);
+    }
+
+    /// @dev Attaches license terms to the given IP on behalf of the owner with a signature.
+    /// @param ipId The ID of the IP to which the license terms will be attached.
+    /// @param licensingModule The address of the Licensing Module.
+    /// @param licenseTemplate The address of the license template.
+    /// @param licenseTermsId The ID of the license terms to be attached.
+    /// @param sigAttach Signature data for attachLicenseTerms to the IP via the Licensing Module.
+    function attachLicenseTermsWithSig(
```

Comment:
internal function should be prefix with "_", e.g.  `function _attachLicenseTermsWithSig(`

Reply:
Thank you for pointing that out!  Since this is an internal library contract function, I believe the underscore convention may not be necessary here. Let me know if you think otherwise!